zend_abstract_interface/exceptions/php7-8/exceptions.c

```diff
@@ -61,19 +83,19 @@ zend_string *zai_get_trace_without_args(zend_array *trace) {
         }
 
         {
-            zval *tmp = zend_hash_find(ht, ZSTR_KNOWN(ZEND_STR_CLASS));
+            zval *tmp = zend_hash_find_ex(ht, ZSTR_KNOWN(ZEND_STR_CLASS), 1);
             if (tmp) {
                 smart_str_appends(&str, Z_TYPE_P(tmp) == IS_STRING ? Z_STRVAL_P(tmp) : "[unknown]");
             }
         }
         {
-            zval *tmp = zend_hash_find(ht, ZSTR_KNOWN(ZEND_STR_TYPE));
+            zval *tmp = zend_hash_find_ex(ht, ZSTR_KNOWN(ZEND_STR_TYPE), 1);
```

Comment:
`_zend_hash_find_known_hash` has been exposed since PHP 7.3, and I like the name better (I had to look up what the `_ex(..., 1)` variant did). I have no idea why it is `ZEND_API` with a leading underscore, but whatever. I think we should just call that instead of going through a conditional parameter to `zend_hash_find_ex`; what do you think?

Reply:
I was pretty sure it to not be ZEND_API :-D I'd be a bit wary of using that one despite the better name … just because it's only used at like 2 places in php-src and as you say underscore prefixed …